dotcom-rendering/src/amp/components/Ad.tsx

```diff
@@ -93,13 +94,27 @@ export const Ad = ({
 	contentType,
 	commercialProperties,
 	rtcConfig,
+	adTargeting,
 }: AdProps) => {
 	const adSizes = isSticky ? stickySizes : inlineSizes;
 	// Set Primary ad size as first element (should be the largest)
 	const [{ width, height }] = adSizes;
 	// Secondary ad sizes
 	const multiSizes = adSizes.map((e) => `${e.width}x${e.height}`).join(',');
 
+	const adTargetingMapped: AdTargetParam[] = [];
+	if (adTargeting.disableAds !== true) {
+		adTargetingMapped.push({
+			name: 'sens',
+			value: String(adTargeting.customParams.sens),
+		});
+
+		adTargetingMapped.push({
+			name: 'urlkw',
+			value: adTargeting.customParams.urlkw as string[],
```

Comment:
It'd be good to avoid the type assertions here since we know what the types are. One option is to update the `CustomParams` type to include more information about the types for the specific keys we know about. Something like:

```typescript
type CustomParams = {
	sens: 't' | 'f';
	urlkw: string[];
	[key: string]: string | string[] | number | number[] | boolean | boolean[];
};
```

So values for `sens` and `urlkw` will have very specific types and any others will have the much looser definition, which should be fine for now. This would avoid the need to call `String(...)` and `as string[]`, but it may require some `as const` assertions in `buildAdTargeting`.

Either way, I think we'll be able to import much more specific types from commercial-core soon!

Reply:
Nice one, thanks for the explanation 👍